packages/rum-core/src/domain/contexts/featureFlagContext.ts

```diff
@@ -0,0 +1,38 @@
+import type { RelativeTime, ContextValue, Context } from '@datadog/browser-core'
+import { assign, isExperimentalFeatureEnabled, SESSION_TIME_OUT_DELAY, ContextHistory } from '@datadog/browser-core'
+import type { LifeCycle } from '../lifeCycle'
+import { LifeCycleEventType } from '../lifeCycle'
+
+export const FEATURE_FLAG_CONTEXT_TIME_OUT_DELAY = SESSION_TIME_OUT_DELAY
+
+export type FeatureFlagContext = Context
+
+export interface FeatureFlagContexts {
+  findFeatureFlagEvaluations: (startTime?: RelativeTime) => FeatureFlagContext | undefined
+  addFeatureFlagEvaluation: (key: string, value: ContextValue) => void
+}
+
+export function startFeatureFlagContexts(lifeCycle: LifeCycle): FeatureFlagContexts {
+  const featureFlagContexts = new ContextHistory<FeatureFlagContext>(FEATURE_FLAG_CONTEXT_TIME_OUT_DELAY)
+
+  lifeCycle.subscribe(LifeCycleEventType.VIEW_ENDED, ({ endClocks }) => {
+    featureFlagContexts.closeActive(endClocks.relative)
+  })
+
+  lifeCycle.subscribe(LifeCycleEventType.VIEW_CREATED, ({ startClocks }) => {
+    featureFlagContexts.add({}, startClocks.relative)
+  })
+
+  return {
+    findFeatureFlagEvaluations: (startTime?: RelativeTime) => featureFlagContexts.find(startTime),
+    addFeatureFlagEvaluation: (key: string, value: ContextValue) => {
+      if (isExperimentalFeatureEnabled('feature_flags')) {
```

Comment:
**💬 suggestion:**
You could check the FF condition first and just return a `noop`  for `findFeatureFlagEvaluations` and `addFeatureFlagEvaluation`. That way you would not need to add any  `lifeCycle.subscribe`